File: 18 Bullseye/javascript/bullseye.js
// BULLSEYE
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//

function print(str)
{
    document.getElementById("output").appendChild(document.createTextNode(str));
}

function input()
{
    var input_element;
    var input_str;
    
    return new Promise(function (resolve) {
                       input_element = document.createElement("INPUT");
                       
                       print("? ");
                       input_element.setAttribute("type", "text");
                       input_element.setAttribute("length", "50");
                       document.getElementById("output").appendChild(input_element);
                       input_element.focus();
                       input_str = undefined;
                       input_element.addEventListener("keydown", function (event) {
                                                      if (event.keyCode == 13) {
                                                      input_str = input_element.value;
                                                      document.getElementById("output").removeChild(input_element);
                                                      print(input_str);
                                                      print("\n");
                                                      resolve(input_str);
                                                      }
                                                      });
                       });
}

function tab(space)
{
    var str = "";
    while (space-- > 0)
        str += " ";
    return str;
}

var as = [];
var s = [];
var w = [];

// Main program
async function main()
{
    print(tab(32) + "BULLSEYE\n");
    print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");
    print("\n");
    print("\n");
    print("\n");
    print("IN THIS GAME, UP TO 20 PLAYERS THROW DARTS AT A TARGET\n");
    print("WITH 10, 20, 30, AND 40 POINT ZONES.  THE OBJECTIVE IS\n");
    print("TO GET 200 POINTS.\n");
    print("\n");
    print("THROW\t\tDESCRIPTION\t\tPROBABLE SCORE\n");
    print("1\t\tFAST OVERARM\t\tBULLSEYE OR COMPLETE MISS\n");
    print("2\t\tCONTROLLED OVERARM\t10, 20 OR 30 POINTS\n");
    print("3\t\tUNDERARM\t\tANYTHING\n");
    print("\n");
    m = 0;
    r = 0;
    for (i = 1; i <= 20; i++)
        s[i] = 0;
    print("HOW MANY PLAYERS");
    n = parseInt(await input());
    print("\n");
    for (i = 1; i <= n; i++) {
        print("NAME OF PLAYER #" + i);
        as[i] = await input();
    }
    do {
        r++;
        print("\n");
        print("ROUND " + r + "\n");
        print("---------\n");
        for (i = 1; i <= n; i++) {
            do {
                print("\n");
                print(as[i] + "'S THROW");
                t = parseInt(await input());
                if (t < 1 || t > 3)
                    print("INPUT 1, 2, OR 3!\n");
            } while (t < 1 || t > 3) ;
            if (t == 1) {
                p1 = 0.65;
                p2 = 0.55;
                p3 = 0.5;
                p4 = 0.5;
            } else if (t == 2) {
                p1 = 0.99;
                p2 = 0.77;
                p3 = 0.43;
                p4 = 0.01;
            } else {
                p1 = 0.95;
                p2 = 0.75;
                p3 = 0.45;
                p4 = 0.05;
            }
            u = Math.random();
            if (u >= p1) {
                print("BULLSEYE!!  40 POINTS!\n");
                b = 40;
            } else if (u >= p2) {
                print("30-POINT ZONE!\n");
                b = 30;
            } else if (u >= p3) {
                print("20-POINT ZONE\n");
                b = 20;
            } else if (u >= p4) {
                print("WHEW!  10 POINT.\n");
                b = 10;
            } else {
                print("MISSED THE TARGET!  TOO BAD.\n");
                b = 0;
            }
            s[i] += b;
            print("TOTAL SCORE = " + s[i] + "\n");
        }
        for (i = 1; i <= n; i++) {
            if (s[i] >= 200) {
                m++;
                w[m] = i;
            }
        }
    } while (m == 0) ;
    print("\n");
    print("WE HAVE A WINNER!!\n");
    print("\n");
    for (i = 1; i <= m; i++)
        print(as[w[i]] + " SCORED " + s[w[i]] + " POINTS.\n");
    print("\n");
    print("THANKS FOR THE GAME.\n");
}

main();

File: 18 Bullseye/perl/bullseye.pl
#!/usr/bin/perl
use strict;


print ' 'x 32 . "BULLSEYE\n";
print ' 'x 15 . "CREATIVE COMPUTING MORRISTOWN, NEW JERSEY\n";
print "\n\n\n";

print "IN THIS GAME, UP TO 20 PLAYERS THROW DARTS AT A TARGET\n";
print "WITH 10, 20, 30, AND 40 POINT ZONES. THE OBJECTIVE IS\n";
print "TO GET 200 POINTS.\n\n";

print "THROW\t\tDESCRIPTION\t\tPROBABLE SCORE\n";
print" 1\t\tFAST OVERARM\t\tBULLSEYE OR COMPLETE MISS\n";
print" 2\t\tCONTROLLED OVERARM\t10, 20 OR 30 POINTS\n";
print" 3\t\tUNDERARM\t\tANYTHING\n\n";

my @A; my $M=0; my $R=0; my @S; my @W; for (my $I=1; $I<=20; $I++) { $S[$I]= 0; }
print "HOW MANY PLAYERS? "; chomp(my $N = <STDIN>); print "\n";
for (my $I=1; $I<=$N; $I++) {
	print "NAME OF PLAYER #$I? "; chomp($A[$I] = <STDIN>);
	}

do {
	$R= $R+1; print "\n"; print "ROUND $R---------\n";
	for (my $I=1; $I<=$N; $I++) {
		my $Flag=0;
		my $T;
		do {
			print "\n"; print "$A[$I]'S THROW? "; chomp($T = <STDIN>);
			if ($T<1 || $T>3) { print "INPUT 1, 2, OR 3!\n"; $Flag=0; }
				else { $Flag=1; }
			} until ($Flag==1);

		my $P1; my $P2; my $P3; my $P4;
		if ($T==1) { $P1=.65; $P2=.55; $P3=.5; $P4=.5; }
		if ($T==2) { $P1=.99; $P2=.77; $P3=.43; $P4=.01; }
		if ($T==3) { $P1=.95; $P2=.75; $P3=.45; $P4=.05; }

		my $B=0;
		my $U= rand(1);
		if ($U>= $P1) { print "BULLSEYE!! 40 POINTS!\n"; $B=40; }
			elsif ($U>= $P2) { print "30-POINT ZONE!\n"; $B=30; }
			elsif ($U>= $P3) { print "20-POINT ZONE\n"; $B=20; }
			elsif ($U>= $P4) { print "WHEW! 10 POINTS.\n"; $B=10; }
			else { print "MISSED THE TARGET! TOO BAD.\n"; $B=0; }
		$S[$I]= $S[$I]+$B; print "TOTAL SCORE =$S[$I]";
		}

	for (my $I=1; $I<=$N; $I++) {
		if ($S[$I]>=200) { $M= $M+1; $W[$M]= $I; } 
		}
	} until ($M!=0);

print "\n"; print "WE HAVE A WINNER!!\n\n";
for (my $I=1; $I<=$M; $I++) { print $A[$W[$I]]." SCORED ".$S[$W[$I]]." POINTS.\n"; }
print "\n"; print "THANKS FOR THE GAME.\n"; exit;


